corehq/apps/hqadmin/reports.py

```diff
@@ -495,3 +502,92 @@ def headers(self):
     @property
     def rows(self):
         return self.table_data.rows
+
+
+class StaleCasesTable:
+    STOP_POINT_DAYS_AGO = 365 * 20
+    AGG_DATE_RANGE = 150
+    STALE_DATE_THRESHOLD_DAYS = 365
+
+    BACKOFF_AMOUNT_DAYS = 30
+    MAX_BACKOFF_COUNT = 2
+
+    def __init__(self):
+        self._rows = None
+        self.stop_date = datetime.now() - timedelta(days=self.STOP_POINT_DAYS_AGO)
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(gettext_lazy("Domain")),
+            DataTablesColumn(gettext_lazy("Case count"))
+        )
+
+    @property
+    def rows(self):
+        if self._rows is None:
+            self._rows = []
+            case_count_by_domain = self._aggregate_case_count_data()
+            for domain, case_count in case_count_by_domain.items():
+                self._rows.append([domain, case_count])
+        return self._rows
+
+    def _aggregate_case_count_data(self):
+        end_date = datetime.now() - timedelta(days=self.STALE_DATE_THRESHOLD_DAYS)
+        agg_res = {}
+        curr_backoff_count = 0
+        curr_agg_date_range = self.AGG_DATE_RANGE
+        domains = self._get_domains()
+        while (True):
+            start_date = end_date - timedelta(days=self.AGG_DATE_RANGE)
+            try:
+                query_res = self._stale_case_count_in_date_range(domains, start_date, end_date)
+            except ESError as e:
+                curr_backoff_count += 1
+                if curr_backoff_count <= self.MAX_BACKOFF_COUNT:
+                    curr_agg_date_range -= self.BACKOFF_AMOUNT_DAYS
+                    curr_backoff_count += 1
+                else:
+                    notify_exception(
+                        None,
+                        'ES query timed out while compiling stale case report email.',
+                        details={
+                            'error': str(e),
+                            'start_date': start_date.strftime("%Y-%m-%d"),
+                            'end_date': end_date.strftime("%Y-%m-%d")
+                        }
+                    )
+                    raise ESError()
+            curr_backoff_count = 0
```

Comment:
I might be wrong here however thie block from here onwards seems like it should be in the else block of try.

Reply:
That is a good point. In my experimental script code I had a continue in the except block which seems I didn't carry over to here. But I like the idea of using an else block rather. Will implement that here.

Reply:
Addressed in [b228b40](https://github.com/dimagi/commcare-hq/pull/35446/commits/b228b408ee63cb87d5786417b375e517c54afcd6)